PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/codeMetrics.ts

```diff
@@ -29,19 +27,72 @@ class CodeMetrics {
     this._consoleWrapper = consoleWrapper
 
     this.normalizeParameters(baseSize, growthRate, testFactor, fileMatchingPatterns, codeFileExtensions)
-
     this.initializeMetrics(gitDiffSummary)
-    this.expectedTestCode = this.metrics.productCode * this.testFactor
-    this.sufficientTestCode = this.metrics.testCode >= this.expectedTestCode
+    this.setSufficientTestCode()
     this.initializeSize()
   }
 
+  private setSufficientTestCode (): void {
+    this._sufficientTestCode = this._metrics.testCode >= (this._metrics.productCode * this._testFactor)
+  }
+
+  public get metrics () {
+    return this._metrics
+  }
+
+  public set metrics (newMetrics: Metrics) {
+    // throw error if input is incorrect
+    this.metrics = newMetrics
+    this.setSufficientTestCode()
+  }
+
+  public get size (): string {
```

Comment:
Most of these should be set internally. You shouldn't need to expose them for the unit tests, so please don't add them